Source/Core/Core/IOS/USB/USBScanner.cpp

```diff
@@ -179,6 +183,26 @@ void USBScanner::AddEmulatedDevices(DeviceMap* new_devices)
   }
 }
 
+void USBScanner::WakeupSantrollerDevice(libusb_device* device) {
+  // Santroller devices emulate various instruments for multiple consoles.
+  // On an actual console, santroller detects the console based on how it communicates
+  // with usb devices. Since the underlying operating system is doing that here, the
+  // check doesn't work, so we need to send a special command to make the device
+  // jump to wii emulation mode.
+  libusb_device_handle* lusb_handle;
+  if (libusb_open(device, &lusb_handle) == LIBUSB_SUCCESS)
+  {
+#ifdef __linux__
+    libusb_set_auto_detach_kernel_driver(lusb_handle, true);
+    libusb_claim_interface(lusb_handle, 2);
+#endif
+    libusb_control_transfer(
+        lusb_handle, +LIBUSB_ENDPOINT_IN | +LIBUSB_REQUEST_TYPE_CLASS | +LIBUSB_RECIPIENT_INTERFACE,
```

Comment:
I don't think you need the three `+`?

Reply:
USBScanner.cpp:201:41: warning: bitwise operation between different enumeration types ‘libusb_endpoint_direction’ and ‘libusb_request_type’ is deprecated [-Wdeprecated-enum-enum-conversion]
  201 |         lusb_handle, LIBUSB_ENDPOINT_IN | LIBUSB_REQUEST_TYPE_CLASS | LIBUSB_RECIPIENT_INTERFACE,
  
 the pluses silence a warning

Reply:
Nah, it's fine like this.